src/framework/auth/services/token/token.ts

```diff
@@ -251,3 +253,48 @@ export class NbAuthOAuth2Token extends NbAuthSimpleToken {
     return JSON.stringify(this.token);
   }
 }
+
+/**
+ * Wrapper for OAuth2 token
+ */
+export class NbAuthOAuth2JWTToken extends NbAuthOAuth2Token {
+
+  static NAME = 'nb:auth:oauth2:jwt:token';
+
+  /**
+   * Returns access token payload
+   * @returns any
+   */
+  getAccessTokenPayload(): any {
+    return decodeJwtPayload(this.getValue())
+  }
+
+  /**
+   * for Oauth2 JWT token, the iat (issued at) field of the access_token payload
+   */
+  protected prepareCreatedAt(date: Date) {
+    let decoded;
+    try {
+       decoded = this.getAccessTokenPayload();
+    }
+    finally {
+      return decoded && decoded.iat ? new Date(Number(decoded.iat) * 1000) : super.prepareCreatedAt(date);
+    }
+  }
+
+  /**
+   * Returns expiration date :
+   * - exp if set,
+   * - super.getExpDate() otherwise
+   * @returns Date
+   */
+  getTokenExpDate(): Date {
+    if (this.getAccessTokenPayload() && this.getAccessTokenPayload().exp) {
```

Comment:
and same here, `exp` is moved into the token payload?
if so, we need to make this `getAccessTokenPayload` call once and probably not cover it with a `null` check same way as for the other tokens

Reply:
Yes exp is present in the token.access-token payload
OK i'll see that